src/peer_sourcehash.cc

```diff
@@ -178,17 +179,13 @@ peerSourceHashSelectParent(PeerSelector *ps)
         score = combined_hash * tp->sourcehash.load_multiplier;
         debugs(39, 3, "peerSourceHashSelectParent: " << tp->name << " combined_hash " << combined_hash  <<
                " score " << std::setprecision(0) << score);
-
-        if ((score > high_score) && peerHTTPOkay(tp, ps)) {
-            p = tp;
-            high_score = score;
+        if (peerHTTPOkay(tp, ps)) {
+            // add negative score we want the highest score first
+            sortedPeers.push_back(std::make_pair(-score, tp));
```

Comment:
Since 987dec6, `addGroup()` performs sorting in descending order. So now, taking that you negate `score`, after `addGroup()` the result would be sorted in ascending order. However, before 987dec6, the peers here were sorted in descending order (you used `std::greater` predicate for that). Is the current behavior correct?

Reply:
please see my comment above.